### PR TITLE
Alerting: Allow pausing alerts from provisioning

### PR DESCRIPTION
**Pull request by alexmobo**
**What is this feature?**

This feature allows alert rules to be paused from file provisioning and API provisioning.

**Why do we need this feature?**

Currently, the pausing alert login is internal to the scheduler but there is no way of pausing alerts from provisioning.

**Who is this feature for?**

All users that use provisioning.

**Which issue(s) does this PR fix?**:

Contributes to https://github.com/grafana/grafana/issues/55519

### CODE REVIEW

File: pkg/services/ngalert/api/tooling/definitions/provisioning_alert_rules.go
Comment (alexweav) [inline pkg/services/ngalert/api/tooling/definitions/provisioning_alert_rules.go:118]: There should also be a change in this function, right now this makes it so the provisioning API can read whether a rule is paused, but the field will not be mapped (and remain false) if something tries to set the paused flag through the prov API.

File: pkg/services/ngalert/api/tooling/definitions/provisioning_alert_rules.go
Comment (alexweav) [inline pkg/services/ngalert/api/tooling/definitions/provisioning_alert_rules.go:114]: Let's make the example `false` for the docs purposes, as a typical rule is not paused

**Comment by JacobsonMT** [comment]
LGTM after the changes mentioned by @alexweav, specifically the change to `UpstreamModel`.

This will conflict with https://github.com/grafana/grafana/pull/58623 in the provisioning markdown docs and the new export models, I can deal with those in my PR once this is merged (or here if this one takes longer than expected).

**Comment by alexmobo** [comment]
I had to add these changes in https://github.com/grafana/grafana/pull/62289 because the `isPaused` field in `AlertRule` model had to be changed to `*bool` to be able to not send the `is_paused` field in `POST /api/ruler/grafana/api/v1/rules/{Namespace}` API call. This means that the default value now is nil and lots of provisioning tests were failing.

**Comment by alexmobo** [comment]
I reopened the ticket as I closed https://github.com/grafana/grafana/pull/62289 and this is still valid.

File: pkg/services/ngalert/schedule/schedule.go
Comment (alexweav) [inline pkg/services/ngalert/schedule/schedule.go:482]: Can this be simplified to `currentRuleVersion > 0 || isPaused`?

File: pkg/services/ngalert/schedule/schedule.go
Comment (alexweav) [inline pkg/services/ngalert/schedule/schedule.go:480]: If a brand new paused rule is file-provisioned (i.e. currentRuleVersion == 0 and isPaused = true), would there be any existing state to begin with? What gets cleaned up?

**Review by alexweav — COMMENTED** [review]
Looking good, couple questions on one of the files but overall looks good!

File: pkg/services/ngalert/schedule/schedule.go
Comment (alexmobo) [reply pkg/services/ngalert/schedule/schedule.go:480]: In that particular case you mention, it won't have any state to clean up but imagine this scenario:
- Have one provisioned alert rule with, at least, one alert instance.
- We update the provisioning file to pause that alert and restart Grafana for these changes to take effect (we must have a DB that lives between restarts for this to happen)

If we don't do this change the previous-restart alert instance from the currently paused alert will linger forever for the user to see until the alert rule is updated (stale states are cleaned up in `ProcessEvalResults` which we skip when an alert is paused).

IMO, a single useless state cleanup is a better price to pay than a forever lingering useless alert instance.

**Review by alexweav — APPROVED** [review]
LGTM

**Review by JacobsonMT — APPROVED** [review]
LGTM

**Pull request merged**
